nilearn/_utils/helpers.py

```diff
@@ -6,7 +6,7 @@
 OPTIONAL_MATPLOTLIB_MIN_VERSION = "3.3.0"
 
 
-def _set_mpl_backend():
+def _set_mpl_backend(message):
```

Comment:
I've added warning as parameter so that we can modify the message depending on the calling package.

Reply:
I've put another warning in `reporting.__init__` file as below:

```
        "nilearn.reporting.make_glm_report and nilearn.reporting.HTMLReport "
        "requires 'matplotlib' to be installed."
        "\nIt can be installed with:\n"
        " pip install 'matplotlib'"
```

The standard message was:
```
    "Some dependencies of nilearn.plotting package seem to be missing."
    "\nThey can be installed with:\n"
    " pip install 'nilearn[plotting]'"
```

I find it more explicit from the users' perspective. It seems strange to me to see the dependency problem for `nilearn.plotting` when I try to import from `nilearn.reporting`.

However I am not sure if reporting requires other dependencies than `matplotlib`. I'll verify that. In either case, I would suggest to customize the warning depending on the origin of the problem.

Finally, if you think that it's ok to return the same message for all, I can revert changes.